### PR TITLE
remove failed genome recovery field dependency

### PR DESCRIPTION
**Pull request by phoenixAja**
### Summary:
- **What:** remove the backend dependency on `czb_failed_genome_recovery` field before removing from the DB
- **Ticket:** **[sc217890](https://app.shortcut.com/genepi/story/217890)**
- **Env:** `<rdev link>`

### Notes:
#1537 needs to be merged in before this

### Checklist:
- [ ] I merged latest `<base branch>`
- [ ] I manually verified the change
- [ ] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/scripts/setup_localdata.py [hunk 1]
```diff
@@ -510,7 +503,7 @@ def create_samples(
     for suffix in range(num_successful):
         _ = create_sample(session, group, user, pathogen, location, suffix)
     for suffix in range(num_failures):
-        _ = create_sample(session, group, user, pathogen, location, suffix, True)
+        _ = create_sample(session, group, user, pathogen, location, suffix)
```
Comment (lvreynoso) [inline src/backend/scripts/setup_localdata.py]: We can probably eliminate this loop, right? And eliminate the `num_failures` parameter in the `create_samples()` function it lives in. We would just have to modify line 565 slightly.

**Review by lvreynoso — APPROVED** [review]
one suggestion, otherwise lgtm

File: src/backend/database_migrations/versions/20230110_175025_remove_czb_failed_genome_recovery_field.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+# revision identifiers, used by Alembic.
+revision = "20230110_175025"
+down_revision = "20230109_223207"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.drop_column("samples", "czb_failed_genome_recovery", schema="aspen")
```
Comment (jgadling) [inline src/backend/database_migrations/versions/20230110_175025_remove_czb_failed_genome_recovery_field.py]: Can we move this migration to a separate PR and deploy it after we've stopped depending on this field?

File: src/backend/scripts/setup_localdata.py (hunk 1 above)
Comment (phoenixAja) [reply src/backend/scripts/setup_localdata.py]: yes thank you

Comment (phoenixAja) [reply src/backend/scripts/setup_localdata.py]: i accidentally missed that

File: src/backend/database_migrations/versions/20230110_175025_remove_czb_failed_genome_recovery_field.py (hunk 2 above)
Comment (phoenixAja) [reply src/backend/database_migrations/versions/20230110_175025_remove_czb_failed_genome_recovery_field.py]: sure

**Review by jgadling — APPROVED** [review]
